### PR TITLE
Word Recommendation Frontend

### PR DESCRIPTION
**Pull request by ESaracay**
Screenshot 2020-08-26 at 10 48 14 PM - Display 2
This PR includes a user interface for the word recommendation servlet. Which allows users to explore words in our new search bar.

### CODE REVIEW

File: frontend/src/main-components/SimilarWordsContent.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    searchParams.append("numOfWordsRequested", this.props.numWords);
+    try {
+      const response = await fetch("/recommendation?" + searchParams.toString())
+        .then(res => res.json());
+      if ("error" in response) {
+        this.setState({ error: true, errorMessage: response.error });
+      }
+      this.setState({ isDataFetched: true, words: response[this.props.word] });
+    } catch (err) {
+      alert("Could not load, refresh page");
```
Comment (zmr) [inline frontend/src/main-components/SimilarWordsContent.js:62]: Nit: "Could not load. Refresh page."

File: frontend/src/pages/CreateCard.js [hunk 2]
```diff
@@ -1,33 +1,31 @@
...
 import Sidebar from "../main-components/Sidebar";
-import styled from "@emotion/styled";
 import CreateCardContent from "../main-components/CreateCardContent";
+import queryString from "query-string";
 
-function CreateCard() {
+function CreateCard(props) {
   // Handles mobile menu button and updates sidebar view
-  const [sideSetting, setSideSetting] = useState("f");
+  const [sideBarVisibility, setSideBarVisibility] = useState(false);
```
Comment (zmr) [inline frontend/src/pages/CreateCard.js:9]: Good to change to true/false rather than use a string

File: frontend/src/pages/CreateCard.js [hunk 3]
```diff
@@ -1,33 +1,31 @@
...
   const handleClick = (e) => {
-    console.log("Clicked");
-    if (sideSetting === "f") {
-      setSideSetting("t");
-    } else {
-      setSideSetting("f");
-    }
+    setSideBarVisibility((sideBarVisibility) => !sideBarVisibility);
```
Comment (zmr) [inline frontend/src/pages/CreateCard.js:11]: as mentioned in another PR, the logic to hide the sidebar is repeated numerous times. Would be good to abstract this out.

**Review by zmr — APPROVED** [review]

File: frontend/src/pages/CreateCard.js (hunk 3 above)
Comment (ESaracay) [reply frontend/src/pages/CreateCard.js:11]: attempted in another PR.

**Pull request merged**